README.md

```diff
@@ -1,11 +1,81 @@
+This template handler will use the AWS CDK to synthesize the CDK stack into a CloudFormation template
+and then run `npx cdk-assets` to publish any required assets to S3/ECR.
+
+**By using the CDK Handler, you are letting CDK synthesize a template, and upload artifacts to S3 and ECR
+and then using Sceptre to actually do the deployment of the template to a stack.**
+In other words, by using this handler with Sceptre, _you skip ever using `cdk deploy`; It's not needed_.
+
+By using this handler, you can now use CDK templates with all your favorite Sceptre commands, like
+`launch`, `validate`, `generate`, and `diff` (along with all the rest)!
+
+## How to install sceptre-cdk-handler
+
+1) Install the Sceptre CDK handler using `pip install sceptre-cdk-handler`
+2) Install the AWS CDK following the instructions in [AWS CDK Getting Started]((https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html))
+3) Install the [CDK-Assets](https://www.npmjs.com/package/cdk-assets) NPM package using `npm install cdk-assets`
+
+## How to use sceptre-cdk-handler
+
+The template "type" for this handler is `cdk`. 
+
+### Arguments:
+
+* `path` (string, required): The path to the CDK template.
+* `class_name` (string, optional): The name of the Class in the CDK template to synthesize. Default: `CdkStack`.
+* `context` (object, optional): The context for the CDK Stack. See [CDK Context](https://docs.aws.amazon.com/cdk/v2/guide/context.html) for further info.
+
+#### CDK Feature Flags and Custom Bootstrap
+
+To set any [CDK Feature Flags](https://docs.aws.amazon.com/cdk/v2/guide/featureflags.html) or to specify a modified
+[CDK Bootstrap](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html) qualifier, set these in the handler's
+`context` argument. See [sceptre-example](sceptre-example) for an example of this.
+
+Reminder: the `context` argument is a standard Sceptre property, so resolvers and/or Jinja variables can be used in the value. 
+
+### How does this handler work?
+
+When using _only_ the CDK CLI (not Sceptre) to deploy using `cdk deploy`, the CDK CLI effectively performs
+the following steps:
+
+1. Synthesises any constructs defined within the CDK Class into a CloudFormation Template and a
+bundle of required assets.
+2. Publishes the assets to a CDK bootstrapped S3 bucket and/or ECR registry.
+3. Creates/updates a CloudFormation stack from the synthesized template.
+
+When you use Sceptre with this handler, the CDK handler performs steps 1-2 above to create a template
+that Sceptre can use, **but it does not use CDK to deploy it!**. Instead, Sceptre can use that template
+produced in step 1 above to perform all it's usual commands with all it's usual magic!
+
+In other words, using this handler lets you use resolvers, put your CDK stack into StackGroups, let
+you name your stack according to Sceptre's naming conventions, `validate`, `diff`, and more! Basically,
+the CDK stack can be managed using Sceptre just like any other.
```

Comment:
I've found that something CDK folks often forget is that you can absolutely use CloudFormation parameters and outputs on CDK stacks. The "typical" CDK structure is to simply interconnect constructs by accessing resources and injecting them into others and such. But parameters and outputs are very well supported (though often forgotten). This is definitely not me asking you to add anything. However, I'm wondering what your thoughts would be on adding a note that you can make CDK stacks more "sceptre-friendly" if they use parameters to receive their input values from other stacks and outputs to send their output values to other stacks.

At least, I'm thinking it might be useful to explain the difference here in how sceptre_user_data is used (template compile-time and can be complex data structures) vs how parameters are used (post-compile, launch-time, need to be simple strings or lists of strings).

Reply:
OK, README and example updated.